When testing for and adding options to CFLAGS/CXXFLAGS, add them at the start
so that local C*FLAGS override them.


--- a/acinclude.m4
+++ b/acinclude.m4
@@ -1728,15 +1728,15 @@
     CFLAGS_saved="$CFLAGS"
     if expr "x$GCC_OPTION" : "x-W.*" >/dev/null
     then
-      CFLAGS="$CFLAGS $ac_wireshark_unknown_warning_option_error $GCC_OPTION"
+      CFLAGS="$GCC_OPTION $CFLAGS $ac_wireshark_unknown_warning_option_error"
     elif expr "x$GCC_OPTION" : "x-f.*" >/dev/null
     then
-      CFLAGS="$CFLAGS -Werror $GCC_OPTION"
+      CFLAGS="$GCC_OPTION $CFLAGS -Werror"
     elif expr "x$GCC_OPTION" : "x-m.*" >/dev/null
     then
-      CFLAGS="$CFLAGS -Werror $GCC_OPTION"
+      CFLAGS="$GCC_OPTION $CFLAGS -Werror"
     else
-      CFLAGS="$CFLAGS $GCC_OPTION"
+      CFLAGS="$GCC_OPTION $CFLAGS"
     fi
     AC_COMPILE_IFELSE(
       [
@@ -1765,7 +1765,7 @@
               # added them, by setting CFLAGS to the saved value plus
               # just the new option.
               #
-              CFLAGS="$CFLAGS_saved $GCC_OPTION"
+              CFLAGS="$GCC_OPTION $CFLAGS_saved"
               #
               # Add it to the flags we use when building build tools.
               #
@@ -1781,7 +1781,7 @@
           # added them, by setting CFLAGS to the saved value plus
           # just the new option.
           #
-          CFLAGS="$CFLAGS_saved $GCC_OPTION"
+          CFLAGS="$GCC_OPTION $CFLAGS_saved"
           #
           # Add it to the flags we use when building build tools.
           #
@@ -1818,15 +1818,15 @@
     CXXFLAGS_saved="$CXXFLAGS"
     if expr "x$GCC_OPTION" : "x-W.*" >/dev/null
     then
-      CXXFLAGS="$CXXFLAGS $ac_wireshark_unknown_warning_option_error $ac_wireshark_non_cxx_warning_option_error $GCC_OPTION"
+      CXXFLAGS="$GCC_OPTION $CXXFLAGS $ac_wireshark_unknown_warning_option_error $ac_wireshark_non_cxx_warning_option_error"
     elif expr "x$GCC_OPTION" : "x-f.*" >/dev/null
     then
-      CXXFLAGS="$CXXFLAGS -Werror $GCC_OPTION"
+      CXXFLAGS="$GCC_OPTION $CXXFLAGS -Werror"
     elif expr "x$GCC_OPTION" : "x-m.*" >/dev/null
     then
-      CXXFLAGS="$CXXFLAGS -Werror $GCC_OPTION"
+      CXXFLAGS="$GCC_OPTION $CXXFLAGS -Werror"
     else
-      CXXFLAGS="$CXXFLAGS $GCC_OPTION"
+      CXXFLAGS="$GCC_OPTION $CXXFLAGS"
     fi
     AC_LANG_PUSH([C++])
     AC_COMPILE_IFELSE(
@@ -1856,7 +1856,7 @@
               # added them, by setting CXXFLAGS to the saved value plus
               # just the new option.
               #
-              CXXFLAGS="$CXXFLAGS_saved $GCC_OPTION"
+              CXXFLAGS="$GCC_OPTION $CXXFLAGS_saved"
             ],
             [
               AC_MSG_RESULT(yes)
@@ -1868,7 +1868,7 @@
           # added them, by setting CXXFLAGS to the saved value plus
           # just the new option.
           #
-          CXXFLAGS="$CXXFLAGS_saved $GCC_OPTION"
+          CXXFLAGS="$GCC_OPTION $CXXFLAGS_saved"
         fi
       ],
       [
